go/pkg/cli/checkout.go

```diff
@@ -188,4 +251,57 @@ func checkoutCheckpoint(opts checkoutOpts, args []string) error {
 `)
 
 	return nil
+
+}
+
+// checkout all the files from an experiment or checkpoint
+func checkoutFileOrDir(outputDir string, checkoutPath string, repo repository.Repository, experiment *project.Experiment, checkpoint *project.Checkpoint) error {
+	// Extract the tarfile
+	experimentFilesExist := true
+	checkpointFilesExist := true
+
+	if err := repo.GetPathItemTar(path.Join("experiments", experiment.ID+".tar.gz"), checkoutPath, outputDir); err != nil {
+		// Ignore does not exist errors
+		if _, ok := err.(*repository.DoesNotExistError); ok {
+			console.Debug("No experiment data found")
+			experimentFilesExist = false
+		} else {
+			return err
+		}
+	} else {
+		console.Info("Copied the files %s from experiment %s to %q", checkoutPath, experiment.ShortID(), filepath.Join(outputDir, experiment.Path))
+	}
+
+	// Overlay checkpoint on top of experiment
+	if checkpoint != nil {
+
+		if err := repo.GetPathItemTar(path.Join("checkpoints", checkpoint.ID+".tar.gz"), checkoutPath, outputDir); err != nil {
+			if _, ok := err.(*repository.DoesNotExistError); ok {
+				console.Debug("No checkpoint data found")
+				checkpointFilesExist = false
+			} else {
+				return err
+
+			}
+		} else {
+			console.Info("Copied the files %s from checkpoint %s to %q", checkoutPath, checkpoint.ShortID(), filepath.Join(outputDir, checkpoint.Path))
```

Comment:
Can we make it "file" if it's a single file or "directory" if it's a directory?

Reply:
Alternatively just "path"

Reply:
Changed to path